jest.environment.js

```diff
@@ -0,0 +1,13 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import Environment from 'jest-environment-jsdom'
+
+export default class CustomTestEnvironment extends Environment {
+  async setup() {
+    await super.setup()
+    this.global.TextEncoder = TextEncoder
+    this.global.TextDecoder = TextDecoder
+    this.global.Response = Response
+    this.global.Request = Request
+    this.global.TransformStream = TransformStream
+  }
+}
```

Comment:
I pulled this solution from stackoverflow https://stackoverflow.com/questions/77358809/referenceerror-textencoder-is-not-defined-during-test-running-with-jest-and-msw

I also needed to include TransformStream. Seems to make sense to me to pass the globals into the environment.